File: engine/src/main/java/ru/vertuos/engine/hedgehog/sonic/SonicMechanics.java
package ru.vertuos.engine.hedgehog.sonic;

import ru.vertuos.engine.hedgehog.HedgehogMechanics;

public class SonicMechanics extends HedgehogMechanics<Sonic, SonicMechanics.SonicSpecialCommand> {

    public enum SonicSpecialCommand {  }

    public SonicMechanics(Sonic hedgehog) {
        super(hedgehog);
    }

    @Override
    public void sendSpecialCommand(SonicSpecialCommand specialCommand) { }
}

File: engine/src/main/kotlin/ru/vertuos/engine/hedgehog/sonic/SonicMechanics.kt
package ru.vertuos.engine.hedgehog.sonic

import ru.vertuos.engine.hedgehog.HedgehogMechanics
import ru.vertuos.engine.hedgehog.sonic.SonicMechanics.SonicSpecialCommand

class SonicMechanics(
    hedgehog: Sonic
) : HedgehogMechanics<Sonic, SonicSpecialCommand>(hedgehog) {

    enum class SonicSpecialCommand

    override fun sendSpecialCommand(specialCommand: SonicSpecialCommand) {

    }
}
